horovod/runner/common/util/hosts.py

```diff
@@ -28,29 +30,21 @@ def from_string(host_string):
         return HostInfo(hostname, int(slots))
 
 
+@dataclass
 class SlotInfo:
-    def __init__(self, hostname, rank, local_rank, cross_rank, size=None, local_size=None, cross_size=None):
```

Comment:
Before, `SlotInfo` creation required `hostname`, `rank`, `local_rank` and `cross_rank`. Now, everything is optional? Does the usage of `SlotInfo` reflect this change?

Reply:
As a [dataclass](https://docs.python.org/3/library/dataclasses.html), because we do not specify defaults, these fields are all required.